RadarSDK/RadarAPIClient.m

```diff
@@ -325,7 +323,9 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
                         if (events && user) {
                             [RadarSettings setId:user._id];
                             
-                            if (!user.trip) {
+                            // if user was on a trip that ended server side, restore previous tracking options
+                            if (!user.trip && [RadarSettings tripOptions]) {
```

Comment:
@corypisano @nickpatrick 

I realize we're building on the existing `if (!user.trip)` logic - but should this be gated behind `if (events)` (line 323) or just  `if (user)`?  I see cases when `events` could be `nil` but we still would want to run this logic.  Maybe in practice events is always at least `[]`and never `nil`?